doc/tutorial/network_ring_gpu.rst

```diff
@@ -0,0 +1,126 @@
+and the MPI communicator for distributed simulations. In some other examples, the :class:`arbor.single_cell_model`
+object created the execution context :class:`arbor.context` behind the scenes. The details of the execution
+context can be customized by the user. We may specify the number of threads in the thread pool; determine the
+id of the GPU to be used; or create our own MPI communicator.
+
+Step **(11)** creates a hardware context where we set the :py:attr:`~arbor.proc_allocation.gpu_id`. This requires 
+that you have built Arbor manually, with GPU support (See :ref:`here <in_python_adv>` how to do that). On a regular 
+consumer device with a single GPU, the index you should pass is ``0``. Change the value to run the example with and 
+without GPU.
+
+.. literalinclude:: ../../python/example/network_ring_gpu.py
+   :language: python
+   :lines: 123-127
+
+Profiling
+*********
+
+Arbor comes with a :class:`arbor.meter_manager` to help you profile your simulations. In this case, you can run the 
+example with ``gpu_id=None`` and ``gpu_id=0`` and observe the difference with the :class:`~arbor.meter_manager`.
+
+Step **(12)** sets up the meter manager and starts it using the (only) context. This way, only Arbor related execution is measured, not Python code.
+
+Step **(13)** instantiates the recipe and sets the first checkpoint on the meter manager. We now have the time it took to construct the recipe.
+
+.. literalinclude:: ../../python/example/network_ring_gpu.py
+   :language: python
+   :lines: 129-136
+
+The domain decomposition
+************************
+
+The domain decomposition describes the distribution of the cells over the available computational resources.
+The :class:`arbor.single_cell_model` also handled that without our knowledge in the previous examples.
+Now, we have to define it ourselves.
+
+The :class:`arbor.domain_decomposition` class can be manually created by the user, by deciding which cells
+go on which ranks. Or we can use a load balancer that can partition the cells across ranks according to
+some rules. Arbor provides :class:`arbor.partition_load_balance`, which, using the recipe and execution
+context, creates the :class:`arbor.domain_decomposition` object for us.
+
+A way to customize :class:`arbor.partition_load_balance` is by providing a :class:`arbor.partition_hint`. They let
+you configure how cells are distributed over the resources in the :class:`~arbor.context`, but without requiring you
+to know the precise configuration of a :class:`~arbor.context` up front. Whether you run your simulation on your 
+laptop CPU, desktop GPU, CPU cluster of GPU cluster, using :class:`partition hints<arbor.partition_hint>` you can
+just say: use GPUs, if available. You only have to change the :class:`~arbor.context` to actually define which 
+hardware Arbor will execute on.
+
+Step **(14)** creates a :class:`arbor.partition_hint`, and tells it to put 1000 cells in a groups allocated to GPUs, 
```

Comment:
I _think_ GPU `p_l_b` prefers to maximise CellGroup sizes, but double check?

Reply:
I don't know what `p_l_b` is, but yes, the 1000 cells is superfluous and purely for showing off the option. I'll add a note.

Reply:
= `partition_load_balance`. Way to long to type out.